src/modExcelToXML.bas

```diff
@@ -0,0 +1,169 @@
+    
+    Dim FSO As New FileSystemObject
+    Dim strBaseName As String
+    Dim strExtension As String
+
+    On Error GoTo errHandler
+    
+    If Right(strPath, Len(strPath) - 1) <> Application.PathSeparator Then
+        strPath = strPath & Application.PathSeparator
+    End If
+    
+    strBaseName = FSO.GetBaseName(strPath & strFileName)
+    strExtension = FSO.GetExtensionName(strPath & strFileName)
+    
+    Call FSO.CopyFile(strPath & strFileName, strPath & strBaseName & ".zip")
+    
+    changeFileExtension = True
+
+Exit Function
+
+errHandler:
+        
+    changeFileExtension = False
+    
+End Function
+
+
+Function windowsUnZip(strUnzipFullPath, strUnzipDestination) As Boolean
+    
+    On Error GoTo errHandler
+    
+    Dim FSO             As FileSystemObject
+    Dim fldr            As Folder
+    Dim objZipApp       As Shell32.Shell
+    
+    Dim strFolderPath   As String
+    Dim strFolderName   As String
+    Dim strZipFolder    As String
+    
+    Set FSO = New FileSystemObject
+    Set objZipApp = New Shell
+    
+    strFolderPath = FSO.GetParentFolderName(strUnzipFullPath)
+    strFolderName = FSO.GetBaseName(strUnzipFullPath)
+    
+    '// TODO this needs to be a function!!!
+    If Right(strFolderPath, Len(strFolderPath) - 1) <> Application.PathSeparator Then
+        strFolderPath = strFolderPath & Application.PathSeparator
+    End If
+    
+    strZipFolder = strFolderPath & strFolderName
+    
+    '// create folder of zip
+    If FSO.FolderExists(strZipFolder) = True Then
+        '// Do nothing
+    Else
+        '// or you can create it
+        Set fldr = FSO.CreateFolder(strZipFolder)
+    
+        If fldr Is Nothing Then
+            MsgBox "Could not create the folder"
+        End If
+    End If
+    
+    With objZipApp
+        .Namespace(strFolderPath & strFolderName).Copyhere .Namespace(strUnzipFullPath).Items
+    End With
+    
+    windowsUnZip = True
+    
+Exit Function
+
+errHandler:
+    
+    windowsUnZip = False
+    
+End Function
+
+
+Function windowsZip(strFilePath, strZipFile)
+  
+  Dim objZipShell As WshShell
+  Dim objZipFSO As FileSystemObject
+  Dim objZipApp As Shell32.Shell
+  Dim lngZipFileCount As Long
+  
+  Dim objFileNameInZip As Object
+  
+  Dim lngLoop As Long
+  
+  Dim strFilePathArray() As String
+  Dim strFileName As String
+  
+  Dim blnDupe As Boolean
+   
+  Set objZipShell = New WshShell
+  Set objZipFSO = New FileSystemObject
+  
+  If Not objZipFSO.FileExists(strZipFile) Then
+    Call newZip(strZipFile)
+  End If
+
+  Set objZipApp = New Shell
+  
+  lngZipFileCount = objZipApp.Namespace(strZipFile).Items.Count
+
+  strFilePathArray = Split(strFilePath, "\")
+  strFileName = (strFilePathArray(UBound(strFilePathArray)))
+  
+  '// listfiles
+  blnDupe = False
+  For Each objFileNameInZip In objZipApp.Namespace(strZipFile).Items
+    If LCase(strFileName) = LCase(objFileNameInZip) Then
+      blnDupe = True
+      Exit For
+    End If
+  Next
```

Comment:
I don't understand why this is done. For CodeExport, wouldn't it be better to just start with a fresh, empty zip file?